### PR TITLE
Update GOVERNANCE.md and MAINTAINERS.md

### PR DESCRIPTION
**Pull request by OrlinVasilev**
- Add new levels of maintainership and SIG deifinitions
- Update MAINTAINERS.md to add SIG Community

Signed-off-by: OrlinVasilev <ovasilev@vmware.com>

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+All the code repositories owned by the Project Contour Github Organization are
+governed by Contour community and maintained under the `ProjectContour` namespace.
+The following repos are the most noteworthy:
 
 * **[Contour](https://github.com/ProjectContour/Contour):** Main Contour codebase.
-* **[Gimbal](https://github.com/ProjectContour/Gimbal):** Gimbal is an ingress load balancing platform capable of routing traffic to multiple Kubernetes and OpenStack clusters
 * **[community](https://github.com/ProjectContour/community):** Used to store community-related material–e.g., proposals, presentation slides, governance documents, community meeting minutes, etc.
-* **[ir2proxy ](https://github.com/ProjectContour/ir2proxy):** ir2proxy is a tool to convert Contour's IngressRoute resources to HTTPProxy resources.
-
+* **[contour-operator](https://github.com/ProjectContour/Contour):** The Contour Operator is a tool for installing and managing Contour.
```
Comment (Miciah) [inline GOVERNANCE.md:19]:
```suggestion
* **[contour-operator](https://github.com/ProjectContour/Contour-Operator):** The Contour Operator is a tool for installing and managing Contour.
```

File: GOVERNANCE.md [hunk 2]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
 
 ## Code Repositories
 
-The following code repositories are governed by Contour community and
-maintained under the `ProjectContour` namespace.
+All the code repositories owned by the Project Contour Github Organization are
```
Comment (Miciah) [inline GOVERNANCE.md:13]:
```suggestion
All the code repositories owned by the Project Contour GitHub Organization are
```

File: GOVERNANCE.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
+**Note:** This document is a work in progress
+
```
Comment (jonasrosland) [inline GOVERNANCE.md:2]: This can be removed. If you want to add "Last updated at DATE" or something similar instead, that'd be ok.

File: GOVERNANCE.md [hunk 4]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
 
 ## Code Repositories
 
-The following code repositories are governed by Contour community and
-maintained under the `ProjectContour` namespace.
+All the code repositories owned by the Project Contour Github Organization are
+governed by Contour community and maintained under the `ProjectContour` namespace.
```
Comment (jonasrosland) [inline GOVERNANCE.md:14]: Why the CamelCase here?

File: GOVERNANCE.md [hunk 5]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+
 * **Users:** Members that engage with the Contour community via any medium (Slack, GitHub, mailing lists, etc.).
 * **Contributors:** Regular contributions to projects (documentation, code reviews, responding to issues, participation in proposal discussions, contributing code, etc.). 
-* **Maintainers**: The Contour project leaders. They are responsible for the overall health and direction of the project; final reviewers of PRs and responsible for releases. Some Maintainers are responsible for one or more components within a project, acting as technical leads for that component. Maintainers are expected to contribute code and documentation, review PRs including ensuring quality of code, triage issues, proactively fix bugs, and perform maintenance tasks for these components.
+* **Reviewer:** Review contributions from other members.
+* **Maintainers**: The Contour project leaders. They are responsible for the overall health and direction of the project; final reviewers of PRs and responsible for releases. 
+* **SIG members/owner**: Members of specific SIG are reposnible for the SIG topic and are marked as maintainers
```
Comment (jonasrosland) [inline GOVERNANCE.md:29]: "Members of a specific SIG are responsible for the work within that SIG topic, and SIG owners are responsible for driving the efforts within a SIG"

File: GOVERNANCE.md [hunk 6]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+* **Reviewer:** Review contributions from other members.
+* **Maintainers**: The Contour project leaders. They are responsible for the overall health and direction of the project; final reviewers of PRs and responsible for releases. 
+* **SIG members/owner**: Members of specific SIG are reposnible for the SIG topic and are marked as maintainers 
+
+
+### User
+WIP: Needs definition
+
+### Contributor
+WIP: Needs definition
```
Comment (jonasrosland) [inline GOVERNANCE.md:36]: Not sure this is needed as they're already clarified above.

File: GOVERNANCE.md [hunk 7]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+### Reviewer
+WIP: Reviewers are able to review code for quality and correctness on some part of a subproject. They are knowledgeable about both the codebase and software engineering principles.
+
+#### Requirements
+
+* member for at least 3 months
```
Comment (jonasrosland) [inline GOVERNANCE.md:42]: "Contributor for at least 3 months"
We don't have project members in the same sense as K8s does.

File: GOVERNANCE.md [hunk 8]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+* Primary reviewer for at least 5 PRs to the codebase
+* Reviewed or merged at least 20 substantial PRs to the codebase
```
Comment (jonasrosland) [inline GOVERNANCE.md:44]: This doesn't really work. They can't be the primary reviewer before they become a reviewer?

File: GOVERNANCE.md [hunk 9]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+* Knowledgeable about the codebase
+* Sponsored by a maintainer
+* With no objections from other maintainer
```
Comment (jonasrosland) [inline GOVERNANCE.md:47]: We should use the same voting system as for a maintainer, super-majority. Not this.

File: GOVERNANCE.md [hunk 10]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+* May either self-nominate, be nominated by an maintainer in this subproject, or be nominated by a robot
```
Comment (jonasrosland) [inline GOVERNANCE.md:48]: Robot? Um...

File: GOVERNANCE.md [hunk 11]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+
+#### Responsibilities and privileges
+
+* Tests are automatically run for PullRequests from members of the Contour GitHub organization
```
Comment (jonasrosland) [inline GOVERNANCE.md:52]: Is this not true for all PRs? I'm unsure, so genuinely asking.

File: GOVERNANCE.md [hunk 12]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+* Assigned test bugs related to subproject of expertise
 
 ### Maintainers
 
 New maintainers must be nominated by an existing maintainer and must be elected by a supermajority of existing maintainers. Likewise, maintainers can be removed by a supermajority of the existing maintainers or can resign by notifying one of the maintainers.
 
+#### Maintainer Expectations
+
+Maintainers have the ability to merge code into the project. Anyone can
+become a Container maintainer (see "Becoming a maintainer" below.)
```
Comment (jonasrosland) [inline GOVERNANCE.md:68]: Where's the "Becoming a maintainer" section?

File: GOVERNANCE.md [hunk 13]
```diff
@@ -8,25 +10,84 @@ This document defines the project governance for Contour.
...
+listed above, they should move themselves to emeritus status. If necessary,
+this can also occur through the decision-making process outlined below.
+
+### SIGs and SIG-owner
+
+SIGs oversee and coordinate the interests and needs of end users and projects in a particular area. By definition, they are long-lived groups that coordinate their work with the maintainers team and are led primarily by recognized experts in the relevant field and supported by other contributors.
+
+SIG Owners are responsible to drive and engage the respective SIG in area of topic. Contribute to code or non-code repositories and sub-projects.
+
+e.g. SIG-Community SIG-Docs...
```
Comment (jonasrosland) [inline GOVERNANCE.md:89]: Remove

File: GOVERNANCE.md [hunk 14]
```diff
@@ -110,3 +171,9 @@ Lazy consensus does _not_ apply to the process of:
 ## Updating Governance
 
 All substantive changes in Governance require a supermajority agreement by all maintainers.
+
+
+## Thanks
```
Comment (jonasrosland) [inline GOVERNANCE.md:176]: Do you want to add in acknowledgements on where some of the new content came from?
Something like "some content in this document were built upon the work in the Kubernetes community" with a link? If that's true. I'm not 100% sure where it comes from :)

File: MAINTAINERS.md [hunk 15]
```diff
@@ -10,7 +10,6 @@ describes governance guidelines and maintainer responsibilities.
 | Nick Young | [youngnick](https://github.com/youngnick) | [VMware](https://www.github.com/vmware/) |
 | Steve Sloka | [stevesloka](https://github.com/stevesloka) | [VMware](https://www.github.com/vmware/) |
 | Steve Kriss | [skriss](https://github.com/skriss) | [VMware](https://www.github.com/vmware/) |
-| Alex Xu |[xaleeks](https://github.com/xaleeks)| [VMware](https://www.github.com/vmware/) |
```
Comment (jonasrosland) [inline MAINTAINERS.md:13]: Removal of a maintainer needs to go through the regular process outlined in the governance doc, so please revert this change.

File: MAINTAINERS.md [hunk 16]
```diff
@@ -20,6 +19,13 @@ describes governance guidelines and maintainer responsibilities.
 * James Peach ([jpeach](https://github.com/jpeach))
 * Daneyon Hansen ([danehans](https://github.com/danehans))
 
+## SIG Community
+
+| Owner | GitHub ID | Affiliation |
+| --------------- | --------- | ----------- |
+| Orlin Vasilev | [OrlinVasilev](https://github.com/OrlinVasilev) | [VMware](https://www.github.com/vmware/) |
+| Alex Xu | [xaleeks](https://github.com/xaleeks) | [VMware](https://www.github.com/vmware/) |
+
```
Comment (jonasrosland) [inline MAINTAINERS.md:28]: Do you want to add SIG Docs here right now and put yourself as owner, and perhaps the technical working group as members?

**Review by jonasrosland — CHANGES_REQUESTED** [review]
Great beginnings here Orlin! Added a bunch of comments :)

File: MAINTAINERS.md (hunk 15 above)
Comment (OrlinVasilev) [reply MAINTAINERS.md:13]: yup already discussed that, will fix

File: GOVERNANCE.md (hunk 3 above)
Comment (OrlinVasilev) [reply GOVERNANCE.md:2]: Actually it's from the k8s gov doc and I liked the idea that it states we are not set in stone but rather ready to update if needed!
https://github.com/kubernetes/community/blob/master/governance.md

we can remove because date will be visible by the last commit!

File: GOVERNANCE.md (hunk 5 above)
Comment (OrlinVasilev) [reply GOVERNANCE.md:29]: yup better!

File: GOVERNANCE.md (hunk 4 above)
Comment (OrlinVasilev) [reply GOVERNANCE.md:12]: I was like that before, I suppose it was for easy reading!

File: GOVERNANCE.md (hunk 12 above)
Comment (OrlinVasilev) [reply GOVERNANCE.md:60]: added!

File: GOVERNANCE.md (hunk 9 above)
Comment (OrlinVasilev) [reply GOVERNANCE.md:39]: done

**Comment by OrlinVasilev** [comment]
@Miciah @jonasrosland all fixed ! :) please check again!

File: GOVERNANCE.md (hunk 9 above)
Comment (skriss) [reply GOVERNANCE.md:36]: I'd remove L36 ("With no objections from other maintainer") since L37 now defines the process.

File: GOVERNANCE.md [hunk 17]
```diff
@@ -8,31 +8,87 @@ This document defines the project governance for Contour.
...
+* Assigned PRs to review related to subproject of expertise
+* Assigned test bugs related to subproject of expertise
 
 ### Maintainers
 
 New maintainers must be nominated by an existing maintainer and must be elected by a supermajority of existing maintainers. Likewise, maintainers can be removed by a supermajority of the existing maintainers or can resign by notifying one of the maintainers.
 
+#### Maintainer Expectations
+
+Maintainers have the ability to merge code into the project. Anyone can
```
Comment (skriss) [inline GOVERNANCE.md:55]: What does "Anyone can become a maintainer" mean specifically? If it's meant to convey that folks from any organization (or no organization at all) are eligible to be maintainers, then I'd say that directly. Otherwise, I'm not sure it really adds anything; the requirements are stated below.

File: GOVERNANCE.md [hunk 18]
```diff
@@ -8,31 +8,87 @@ This document defines the project governance for Contour.
...
+
+### Becoming a maintainer
+
+* Anyone can become a Contour maintainer
+* Maintainers should be proficient in Go
+* have relevant domain expertise
+* have the time and ability to meet the maintainer expectations above
+* and demonstrate the ability to work with the existing maintainers and project processes
+
+*New maintainers must be nominated by an existing maintainer and must be elected by a supermajority of existing maintainers. Likewise, maintainers can be removed by a supermajority of the existing maintainers or can resign by notifying one of the maintainers.*
```
Comment (skriss) [inline GOVERNANCE.md:90]: Should we move this under the Maintainers section, above?

File: GOVERNANCE.md [hunk 19]
```diff
@@ -8,31 +8,87 @@ This document defines the project governance for Contour.
...
 ### Supermajority
 
 A supermajority is defined as two-thirds of members in the group.
 A supermajority of [Maintainers](#maintainers) is required for certain
 decisions as outlined above. Voting on decisions can happen on the mailing list, GitHub, Slack, email, or via a voting service, when appropriate. Maintainers can either vote "agree, yes, +1", "disagree, no, -1", or "abstain". A vote passes when supermajority is met. An abstain vote equals not voting at all.
 
+
+### Becoming a maintainer
+
+* Anyone can become a Contour maintainer
```
Comment (skriss) [inline GOVERNANCE.md:84]: same question about what this statement is trying to convey, since clearly there are requirements

**Review by skriss — COMMENTED** [review]
Couple comments but generally :+1: to the additional role definitions

File: GOVERNANCE.md (hunk 9 above)
Comment (youngnick) [reply GOVERNANCE.md:36]: Yeah, I think L36 is now redundant.

File: GOVERNANCE.md (hunk 18 above)
Comment (youngnick) [reply GOVERNANCE.md:90]: Yeah, feels like this should be a subsection of the Maintainers heading.

**Review by youngnick — COMMENTED** [review]
This is mostly LGTM, with just a few small things.

**Comment by youngnick** [comment]
Ping @OrlinVasilev, this is looking pretty close to me, if you can resolve the outstanding comments I think we can get it in and have the SIGs officially set up. That should make your CNCF interactions a little easier.

**Comment by OrlinVasilev** [comment]
@youngnick @skriss All fixed!

**Review by skriss — APPROVED** [review]

**Review by youngnick — APPROVED** [review]
This is great, thanks @OrlinVasilev! I'll merge now, and we can announce the SIG formation in this week's meeting.

**Comment by OrlinVasilev** [comment]
Great thank you @youngnick !!!
